Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt

```diff
@@ -108,6 +110,7 @@ class MainActivity : AppCompatActivity(), HasAndroidInjector {
         scheduleWork()
         vm.doBackgroundNoiseCheck()
         contactDiaryWorkScheduler.schedulePeriodic()
+        dataDonationAnalyticsScheduler.schedulePeriodic()
```

Comment:
Adding this here will not schedule new worker does it?
The logs are misleading:
```
2021-02-22 15:15:12.896 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 8 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:18:40.306 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 8 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:18:57.219 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 18 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:19:15.002 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 1 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:19:21.499 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 21 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:19:27.070 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 6 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:19:55.359 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 1 Hours in the future, will repeat every 24 Hours from there
2021-02-24 15:20:29.996 25009-25009/de.rki.coronawarnapp.test D/DataDonationAnalyticsScheduler: scheduling Analytics job for 0 Hours in the future, will repeat every 24 Hours from there
```

While actually the first worker was scheduled and kept check the timestamps for the first log and the start time in the screenshot:
<img width="1507" alt="Screenshot 2021-02-22 at 15 24 31" src="https://user-images.githubusercontent.com/25054729/108721315-2977ff80-7522-11eb-8b10-a65e05886e14.png">


my question here why it is moved from Application into here ?

Reply:
It is moved here because on application init the Initial onboarding might not be completed and there is no need to schedule a job that will never be able to execute, also onboarding might complete after app initialisation (first launch) which would not have scheduled a job before

The logging is quite misleading yes, since we enqueue the work using `ExistingPeriodicWorkPolicy.KEEP` only the first ever call to `schedulePeriodic` will actually enqueue a task, sadly there is (afaik) no way for us to know if a task has been queued before so there is no way to make the logging more usefull

Reply:
maybe removing the log completely would be a solution, but i don't really like the idea

Reply:
Ok, I checked `enqueueUniquePeriodicWork` which return Operation status, but it does not say whether it was enqueued before or not. Maybe just rephrase the log message ,so it gives the reader a hint.
such as : "Scheduling Analytics job with KEEP policy for %s Hours in the future, will repeat every 24 Hours from there. Won't schedule if is already enqueued" or whatever you like 😃

Reply:
As @d4rken has also added logging here to his pr i will remove mine

Reply:
see: https://github.com/corona-warn-app/cwa-app-android/pull/2436